models/service.go

```diff
@@ -39,6 +45,44 @@ func (s *ServiceList) SetServiceList(serviceList *kubernetes.ServiceList) {
 	s.Services = CastServiceOverviewCollection(serviceList)
 }
 
+// Aggregates requests counts from metrics fetched from Prometheus,
+// calculates error rates for each service and stores the result in the service list.
+func (s *ServiceList) ProcessRequestCounters(requestCounters prometheus.MetricsVector) {
+	// First, aggregate request counters (both in and out)
+	for _, sample := range requestCounters.Vector {
+		serviceDst := strings.SplitN(string(sample.Metric["destination_service"]), ".", 2)[0]
+		serviceSrc := strings.SplitN(string(sample.Metric["source_service"]), ".", 2)[0]
+		servicesFound := 0
+
+		for idx := range s.Services {
+			service := &s.Services[idx]
```

Comment:
Why not directly `for _, service := range s.Services` ?

Reply:
I tried. But using that syntax `service` holds a copy, so doing `service.RequestCount += sample.Value` will leave untouched the elements in the array. Even, this is why I'm using `service := &s.Services[idx]` (with the ampersand)

Reply:
ok!